lib/mix/lib/mix/dep/lock.ex

```diff
@@ -62,4 +63,11 @@ defmodule Mix.Dep.Lock do
   defp lockfile do
     Mix.Project.config[:lockfile]
   end
+
+  defp assert_no_merge_conflicts_in_lockfile(lockfile, info) do
+    if info =~ ~r/^(?:<<<<<<<|=======|>>>>>>>|\|\|\|\|\|\|\|)/ do
+      Mix.raise "Your #{lockfile} contains merge conflicts. If you are using git, " <>
+                "run `git checkout HEAD -- #{lockfile}` to revert local changes"
```

Comment:
I agree on this. Maybe we can just say "fix the merge conflicts" or something along these lines.